src/main/java/org/icatproject/core/entity/DataPublication.java

```diff
@@ -0,0 +1,157 @@
+package org.icatproject.core.entity;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
+import javax.persistence.UniqueConstraint;
+
+@Comment("A curated data publication")
+@SuppressWarnings("serial")
+@Entity
+@Table(uniqueConstraints = { @UniqueConstraint(columnNames = { "FACILITY_ID", "PID" }) })
+public class DataPublication extends EntityBaseBean implements Serializable {
+
+	@JoinColumn(name = "FACILITY_ID", nullable = false)
+	@ManyToOne(fetch = FetchType.LAZY)
+	private Facility facility;
+
+	@JoinColumn(name = "DATACOLLECTION_ID", nullable = false)
+	@ManyToOne(fetch = FetchType.LAZY)
+	private DataCollection content;
+
+	@OneToMany(cascade = CascadeType.ALL, mappedBy = "publication")
+	private List<DataPublicationUser> users = new ArrayList<>();
+
+	@OneToMany(cascade = CascadeType.ALL, mappedBy = "publication")
+	private List<DataPublicationDate> dates = new ArrayList<>();
+
+	@OneToMany(cascade = CascadeType.ALL, mappedBy = "publication")
+	private List<RelatedIdentifier> relatedIdentifiers = new ArrayList<>();
+
+	@OneToMany(cascade = CascadeType.ALL, mappedBy = "dataPublication")
+	private List<DataPublicationFunding> fundingReferences = new ArrayList<>();
```

Comment:
I believe the abbreviated attribute names (`publication` to refer to a `DataPublication`) are justified here.

The new entity classes `DataPublicationDate`, `DataPublicationUser`, and `RelatedIdentifier` are centered around `DataPublication`.  Objects in these classes are entirely dependent on the publication they relate to, they are meaningless without it.  It is obvious from this context that an attribute named `publication` in these classes can only refer to the `DataPublication`.  Overly verbose attribute names do not improve readability of the code.  This goes in the same way as the `type` attribute in `Investigation` refers to the `InvestigationType` and not to a `SampleType` and the `parameters` in `Dataset` refer to `DatasetParameters`.

The situation is slightly different for `DataPublicationFunding`, because that classes establishes the relation between a `DataPublication` and a `FundingReference`.  A `FundingReference` object does have a meaningful existence independent from `DataPublication` in the new ICAT schema.

So I'd prefer leaving the `publication` attributes in `DataPublicationDate`, `DataPublicationUser`, and `RelatedIdentifier`.

Reply:
OK. If you're comfortable with it I don't feel strongly enough about it to argue the case for it to be changed.